### DISCUSSION TITLE
IP Geolocation Default or Maxmind

### DISCUSSION THREAD

**Discussion by Piskatje13** [plausible/analytics#4150] [Self-Hosted Support]
Hi,

1. Which IP Geolocation do they use in https://plausible.io ?
2. I need IP Geolocation per city. Which one is the best?
3. Do I need to install the default Geolocation? Will it work for cities?

Thank you in advance!

**Comment by ruslandoga** [comment]
👋 @Piskatje13 

Default geolocation DB (from db-ip) doesn't have geoname ids for cities so it cannot be used for city-level geolocation. Please see https://github.com/plausible/community-edition/tree/v2.1.0?tab=readme-ov-file#ip-geolocation for how to setup MaxMind for city-level geolocation.

**Comment by Piskatje13** [comment]
Thank you! How about my question 1? Which IP Geolocation do they use in https://plausible.io/ ?

**Reply by ruslandoga** [reply]
I don't know.

**Comment by Piskatje13** [comment]
I have been testing MaxMind with self-hosted Matomo. We need statistics for our city. MaxMind shows 80-90% (!) less visitors in our city compared to GA, Clarity or plausible.io (I have a trial). I do not want to use GA and Clarity due the cookies tracking.

**Reply by ruslandoga** [reply]
I think you should try using MaxMind with Plausible first :)

**Comment by Piskatje13** [comment]
I will do!
PS: I got a response from the helpdesk:
_We use the [MaxMind database](https://www.maxmind.com/) to determine the visitor location based on the IP address. This lookup happens on our servers and no data is ever sent to MaxMind._

**Comment by Piskatje13** [comment]
This is my setting plausible-conf.env

ADMIN_USER_EMAIL=gxx@gmail.com
ADMIN_USER_NAME=xx
ADMIN_USER_PWD=xx
BASE_URL=https://website.com
SECRET_KEY_BASE=kxxxxxxx
TOTP_VAULT_KEY=exxxxx

MAILER_NAME=Plausible
MAILER_EMAIL=analytics@email.com
SMTP_HOST_ADDR=web106.xxxx.net
SMTP_HOST_PORT=465
SMTP_USER_NAME=analytics@email.com
SMTP_USER_PWD=xxx
GOOGLE_CLIENT_ID=3xxxxxxx.apps.googleusercontent.com 
GOOGLE_CLIENT_SECRET=xxxxx

when I add

MAXMIND_LICENSE_KEY=MY-KEY
MAXMIND_EDITION=GeoLite2-City

My Plausible goes down... Why?

**Reply by ruslandoga** [reply]
City-level MMDB uses more RAM. That could be the reason.

**Comment by Piskatje13** [comment]
Successfully added MAXMIND. My Plausible is online now. Beautiful! But the statistics for Countries, Regions and Cities are empty:( Is there a way how I check the problem?

**Reply by ruslandoga** [reply]
Yes, you can start an interactive sessions and lookup some IP addresses manually as a check:

```console
$ docker compose exec plausible bin/plausible remote
```
```elixir
iex> Plausible.Geo.lookup("3.2.3.4") # or use your own IP address
```

**Reply by ruslandoga** [reply]
You can also check what geoname IDs were persisted for events with a query to ClickHouse:

```console
$ docker compose exec plausible_events_db clickhouse client --database plausible_events_db
```
```sql
:) select timestamp, name, hostname, country_code, subdivision1_code, subdivision2_code, city_geoname_id from events_v2 order by timestamp desc limit 10;
```

**Comment by Piskatje13** [comment]
Thanks! It works now. How often does Plausible download the MAXMIND database?

**Reply by ruslandoga** [reply]
Plausible uses `:locus` library so it updates remote databases using this [this approach.](https://github.com/g-andrade/locus?tab=readme-ov-file#remote-sources-downloading-and-updating)

**Comment by abdullahdevrel** [comment]
I have added a feature request to integrate IPinfo's IP to Country database: https://feedback.plausible.io/502.

https://ipinfo.io/products/free-ip-database

It has a superior advantage over the existing solution as it provides full accuracy, daily updates, and downloading does not require user tokens as licensing terms.

IPinfo is also supported by the existing Locus library: https://github.com/g-andrade/locus?tab=readme-ov-file#alternative-providers.

Feel free to explore it. If you have any questions let me know.

**Reply by ruslandoga** [reply]
Discussion moved to https://github.com/plausible/community-edition/issues/170
